adl/stdlib/sys/annotations.adl

```diff
@@ -19,4 +19,81 @@ struct SerializedWithInternalTag {
 
 type CustomSerialization = Bool;
 
+
+// Experimental annotation to allow the lifting of struct fields to unions.
+// That is a backward compatiblity feature.
+// In effect(*) the deserializer will try to deserialize into the union first,
+// is this fails it will try deserializing into a type of the first branch.
```

Comment:
I think the correct behaviour is to

1) try to deserialize into the union
2) if 1 fails then try the type of the first branch

But if (2) fails, then we should throw the error originally thrown by (1).